feat: per-message thumbs up/down feedback (#9718)
Kilo-Org/kilocode: packages/kilo-docs/pages/code-with-ai/features/message-feedback.md

---
title: "Message Feedback"
description: "Rate assistant responses with thumbs up/down to help Kilo improve"
---

# Message Feedback

You can give a thumbs up or thumbs down to any response Kilo gives you. It's a quick way to tell us when something worked well, or when it didn't.

{% callout type="info" %}
Feedback is only available when telemetry is on. If you turn telemetry off, the buttons no longer display. See [Turning it off](#turning-it-off) below.
{% /callout %}

## How to use it

{% tabs %}
{% tab label="VSCode" %}

When Kilo finishes a response, you'll see thumbs-up and thumbs-down buttons next to the **Copy response** button. Click one to send your rating. Click the same button again to take it back, or click the other one to change your mind.

> If you reload the window or switch to a different session, the buttons will look unrated again — but don't worry, your earlier ratings were already sent.

Author: ```suggestion
If you reload the window or switch to a different session, the feedback status will be reset.
```
